include/device.h

```diff
@@ -344,6 +397,78 @@ struct device {
 #endif
 };
 
+/**
+ * @brief Get the handle for a given device
+ *
+ * @param dev the device for which a handle is desired.
+ *
+ * @return the handle for the device.
+ */
+static inline device_handle_t
+device_get_handle(const struct device *dev)
+{
+	extern const struct device __device_start[];
+
+	/* TODO: If/when devices can be constructed that are not part of the
+	 * fixed we'll need another solution.
+	 */
+	return (device_handle_t)(dev - __device_start);
+}
+
+/**
+ * @brief Get the device corresponding to a handle.
+ *
+ * @param dev_handle the device handle
+ *
+ * @return the device that has that handle, or a null pointer if @p
+ * dev_handle does not identify a device.
+ */
+static inline const struct device *
+device_from_handle(device_handle_t dev_handle)
```

Comment:
`device_from_handle_get()` maybe

Reply:
`device from handle get` is pretty contorted syntax. But let's see what others think.

Reply:
DT APIs use "by", e.g. `DT_PHA_BY_IDX()`, `DT_PHA_BY_NAME()`, which could point to `device_get_by_handle()` or `device_by_handle()`.

That still looks weird to my US English eyes, though, in a way that `device_get_from_handle()` does not.